packages/@react-spectrum/combobox/test/ComboBox.test.js

```diff
@@ -0,0 +1,1486 @@
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        act(() => {
+          combobox.focus();
+          jest.runAllTimers();
+        });
+        expect(onOpenChange).toHaveBeenLastCalledWith(true);
+
+        let listbox = getByRole('listbox');
+        expect(onOpenChange).toBeCalledTimes(1);
+        expect(onOpenChange).toHaveBeenCalledWith(true);
+        testComboBoxOpen(combobox, button, listbox, 0);
+      });
+    });
+
+    describe('via isOpen and defaultOpen', function () {
+      it('has a controlled open state via isOpen', function () {
+        let {getByRole} = renderComboBox({isOpen: true});
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        expect(() => getByRole('listbox')).toThrow();
+
+        act(() => {
+          combobox.focus();
+          jest.runAllTimers();
+        });
+
+        let listbox = getByRole('listbox');
+        expect(onOpenChange).toBeCalledTimes(0);
+        testComboBoxOpen(combobox, button, listbox, 0);
+
+        act(() => {
+          triggerPress(button);
+          jest.runAllTimers();
+        });
+
+        expect(listbox).toBeVisible();
+        expect(onOpenChange).toBeCalledTimes(1);
+        expect(onOpenChange).toHaveBeenCalledWith(false);
+      });
+
+      it('has a uncontrolled open state via defaultOpen', function () {
+        let {getByRole} = renderComboBox({defaultOpen: true});
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        expect(() => getByRole('listbox')).toThrow();
+
+        act(() => {
+          combobox.focus();
+          jest.runAllTimers();
+        });
+
+        let listbox = getByRole('listbox');
+        expect(onOpenChange).toBeCalledTimes(0);
+        testComboBoxOpen(combobox, button, listbox, 0);
+
+        act(() => {
+          triggerPress(button);
+          jest.runAllTimers();
+        });
+
+        expect(() => getByRole('listbox')).toThrow();
+        expect(onOpenChange).toBeCalledTimes(1);
+        expect(onOpenChange).toHaveBeenCalledWith(false);
+        expect(button).toHaveAttribute('aria-expanded', 'false');
+        expect(button).not.toHaveAttribute('aria-controls');
+        expect(combobox).not.toHaveAttribute('aria-controls');
+      });
+    });
+
+    describe('button click', function () {
+      it('keeps focus within the textfield after opening the menu', function () {
+        let {getByRole} = renderComboBox();
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        expect(() => getByRole('listbox')).toThrow();
+
+        act(() => {
+          combobox.focus();
+          triggerPress(button);
+          jest.runAllTimers();
+        });
+
+        let listbox = getByRole('listbox');
+        expect(listbox).toBeTruthy;
+        expect(document.activeElement).toBe(combobox);
+      });
+
+      // Double check this test
```

Comment:
has this been double checked?

Reply:
Waiting to see is onFilter is implemented correctly in this PR